rust-src/id/src/ffi.rs

```diff
@@ -292,6 +299,274 @@ pub extern "C" fn ip_info_cdi_verify_key(
     ptr
 }
 
+/// Attempt to create an [`ArInfo`] instance from byte array pointers.
+/// Assumes that public_key_ptr points to a PublicKey<G1>
+/// instance and that name_ptr, url_ptr and desc_ptr
+/// to valid utf8 strings, and returns a null-pointer
+/// otherwise.
+#[no_mangle]
+unsafe extern "C" fn ar_info_create(
+    identity: u32,
+    public_key_ptr: *mut u8,
+    public_key_len: size_t,
+    name_ptr: *const u8,
+    name_len: size_t,
+    url_ptr: *const u8,
+    url_len: size_t,
+    desc_ptr: *const u8,
+    desc_len: size_t,
+) -> *mut ArInfo<G1> {
+    if let Ok(ptr) = ar_info_create_helper(
+        identity,
+        public_key_ptr,
+        public_key_len,
+        name_ptr,
+        name_len,
+        url_ptr,
+        url_len,
+        desc_ptr,
+        desc_len,
+    ) {
+        return ptr;
+    }
+    std::ptr::null_mut()
```

Comment:
```suggestion
    ) {
        ptr
    } else {
        std::ptr::null_mut()
    }
```
Rust is expression based. Please use that.